### PR TITLE
Separate module file for all challenge generation/validation

### PR DESCRIPTION
**Pull request by akapur99**
Small PR just to prevent blockage on further tickets such as #340
Spinoff of PR #11 dealing with ticket #339 : Handles generation of challenge string

Related future work will be addressed in following ticket:
#354 End-to-end BDD Testing for Auth-Service Endpoints

### CODE REVIEW

**Comment by spikerheado1234** [comment]
Lgtm!

File: src/lib/challenge.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Helper function which uses the crypto library to generate
+ * a secure random challenge string (Base 64 encoding) of given size
+ * @param size Integer value of how many bytes should generated, 32 by default
+ */
+export async function generate (size: number = 32): Promise<string> {
+  try {
+    const buf = await randomBytesAsync(Math.round(Math.abs(size)))
+    return buf.toString('base64')
+  } catch (error) {
+    Logger.error('Unable to generate challenge string')
```
Comment (eoln) [inline src/lib/challenge.ts:47]:
```suggestion
    Logger.push(error).error('Unable to generate challenge string')
```

File: test/unit/lib/challenge.test.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { generate } from '../../../src/lib/challenge'
+
+describe('Challenge Generation', (): void => {
+  it('Should return a 32 byte string', async (): Promise<void> => {
```
Comment (eoln) [inline test/unit/lib/challenge.test.ts:4]:
```suggestion
  it('Should return a 32 byte string by default', async (): Promise<void> => {
```

File: test/unit/lib/challenge.test.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    const challenge = await generate()
+    expect(Buffer.byteLength(challenge, 'base64')).toBe(32)
+  })
+
+  it('Should return a 64 byte string', async (): Promise<void> => {
```
Comment (eoln) [inline test/unit/lib/challenge.test.ts:9]:
```suggestion
  it('Should return a 64 byte string if argument passed', async (): Promise<void> => {
```

File: test/unit/lib/challenge.test.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { generate } from '../../../src/lib/challenge'
```
Comment (eoln) [inline test/unit/lib/challenge.test.ts:1]: there should be cases when `size` parameter is nonpositive number

**Pull request merged**